src/common/libsubprocess/subprocess.c

```diff
@@ -246,20 +246,23 @@ void subprocess_standard_output (flux_subprocess_t *p, const char *stream)
 
 void subprocess_check_completed (flux_subprocess_t *p)
 {
-    assert (p->state == FLUX_SUBPROCESS_EXITED);
-
-    /* we're also waiting for the "complete" to come from the remote end */
-    if (!p->local && !p->remote_completed)
-        return;
+    if (p->state == FLUX_SUBPROCESS_EXITED) {
```

Comment:
Issue # is blank in commit message.

It would be easier to view the diff if the `assert()` were changed to:
```c
    if (p->state != FLUX_SUBPROCESS_EXITED) {
        log_err ("subprocess_check_completed: unexpected state %s",
                 flux_subprocess_state_string (p->state));
        return;
    }
```

However, that's just personal preference and perhaps it is more correct to do it the way is done here, so feel free to ignore that comment.